src/components/action-popover/action-popover.spec.js

```diff
@@ -888,4 +899,33 @@ describe('ActionPopover', () => {
       global.console.error.mockReset();
     });
   });
+
+  describe('Custom Menu Button', () => {
+    it('supports being passed an override component to act as the menu button', () => {
+      const popover = enzymeMount(
+        <ThemeProvider theme={ mintTheme }>
+          <ActionPopover renderButton={ props => <Button { ...props }>Foo</Button> }>
+            <ActionPopoverItem onClick={ jest.fn() }>foo</ActionPopoverItem>
+          </ActionPopover>
+        </ThemeProvider>
+      ).find(ActionPopover);
+
+      const button = popover.find(Button);
+
+      expect(popover.find(Icon).first().exists()).toBeFalsy();
+      expect(button.exists()).toBeTruthy();
+      expect(button.props().tabIndex).toEqual(-1);
+      expect(button.props()['data-element']).toEqual('action-popover-menu-button');
```

Comment:
Would this be a good candidate for a snapshot test? Seems to be a case of "given a renderButton prop I expect the markup to be like this".

Reply:
I considered it here but I had to add a fair few snapshots when I was doing the submenu work and felt that adding another might be excessive. I decided against it here as I found an alternative way of achieving the same assertion but change it to a snapshot if the consensus is that it's preferred